### PR TITLE
TINYGL: reduce code duplication in ztriangles (was: [RFC] TINYGL: unroll some loops)

### PR DESCRIPTION
**Pull request by vpelletier**
Some loops are written in a way which suggests loop-unrolling was intended, but code was re-rolled. This change unrolls these loops.

Note: I did not see any obvious performance difference. If anything, this makes reading the code easier by reducing repetitions, and maybe fixing tiny (runs of 3 pixels) rendering approximations.

This brings #1270 changes because of conflicts.

### CODE REVIEW

**Comment by vpelletier** [comment]
...and this actually decreases performance. Funny.

**Comment by vpelletier** [comment]
With this last patch, an initial 62FPS in first EMI scene (without this pull request) is increased to 65 FPS (with unrolling disabled) instead of decreasing to 58FPS.

Should I track more precisely which unrolls are beneficial and detrimental ?
Once the unrolls to drop are known, should I keep them around or just delete them ?

**Comment by vpelletier** [comment]
The commit disabling manual loop unrolling breaks MYST 3 (in a rather funny, wobbly way).

**Comment by vpelletier** [comment]
And the regression is what caused the performance improvement: in the `(kInterpST || kInterpSTZ)` branch, s, t, dsdx, dtdx and fz were not updated at all, causing the texture rendering glitches. Re-enabling their computation, the speed is at 62FPS without any unrolling (so overall, manual unrolling still costs 4 FPS).
I fix this in latest version of this PR.
Maybe I should split the second commit further ?

**Comment by vpelletier** [comment]
FWIW, I am scavenging commits which do not add 32bits support per-se from this pull request (and other changes I initially prepared after that branch), so that they can be merged independently and I can keep this branch floating over master until the surrounding situation makes a merge acceptable.

**Comment by vpelletier** [comment]
Now this PR only contains a single commit, which does not depent on 32bits work but mostly reduces code duplication (which in turn makes 32bits work a much simpler patch).

File: graphics/tinygl/ztriangle.cpp [hunk 1]
```diff
@@ -403,6 +403,8 @@ void FrameBuffer::fillTriangle(ZBufferPoint *p0, ZBufferPoint *p1, ZBufferPoint
 			{
 				if (kDrawLogic == DRAW_DEPTH_ONLY ||
 						(kDrawLogic == DRAW_FLAT && !(kInterpST || kInterpSTZ))) {
+#define UNROLL_putPixelDepth(_a) putPixelDepth<kDepthWrite, kEnableScissor>(this, buf, pz, _a, x, y, z, dzdx)
```
Comment (aquadran) [inline graphics/tinygl/ztriangle.cpp]: No macros please :)

**Comment by vpelletier** [comment]
Now without macros.

**Pull request merged**